x/gamm/types/pool_asset.go

```diff
@@ -107,3 +107,12 @@ func PoolAssetsCoins(assets []PoolAsset) sdk.Coins {
 	}
 	return coins
 }
+
+func GetPoolAssetByDenom(assets []PoolAsset, denom string) (PoolAsset, bool) {
+	for _, asset := range assets {
+		if asset.Token.Denom == denom {
+			return asset, true
+		}
+	}
+	return PoolAsset{}, false
```

Comment:
If we go with the suggestion above, could return nil here instead of an empty `PoolAsset`